src/utils/EntityFactory.ts

```diff
@@ -25,7 +47,10 @@ export async function buildEntityAndFile(
   }
 
   const entityFile = Buffer.from(JSON.stringify(entity))
-  const entityId: EntityId = await Hashing.calculateBufferHash(entityFile)
+  const entityId: EntityId =
+    version === EntityVersion.V3
```

Comment:
Shouldn't we ask for v4 instead? As we are sure that it's present in the new version

Reply:
I don't understand the comment. The idea is that if the version is v3, we will use the current hashing method. If the version is v4 or higher, we will use ipfs